code/game/atoms.dm

```diff
@@ -679,9 +679,13 @@
   * Default behaviour is to call [contents_explosion][/atom/proc/contents_explosion] and send the [COMSIG_ATOM_EX_ACT] signal
   */
 /atom/proc/ex_act(severity, target)
+	SHOULD_CALL_PARENT(TRUE)
 	set waitfor = FALSE
+	if(QDELETED(src))
+		return
+	if(SEND_SIGNAL(src, COMSIG_ATOM_EX_ACT, severity, target) & COMPONENT_BLOCK_EX_ACT)
+		return
```

Comment:
```suggestion
	if(SEND_SIGNAL(src, COMSIG_ATOM_EX_ACT, severity, target) & COMPONENT_BLOCK_EX_ACT)
		return TRUE
```
If you think this request is too much let me know, but ideally things call parent first and check the return of the signal, instead of deleting themselves regardless before calling parent.